### DISCUSSION TITLE
Is it possible to replace a sound file of AudioClip asset?

### DISCUSSION THREAD

**Discussion by Tudzer** [nesrak1/UABEA#281] [Q&A]
I need to replace some voiceover sound files, but newer files are bigger so it would require some smart batch patching of assets' offsets as `resources.resource` file would change in size. Or maybe there is some alternative way to make game that doesn't use `StreamingAssets` to use it and overwrite any existing assets that match by name with those from `StreamingAssets`  folder?

**Comment by eternalfuture-e38299** [comment]
The audio can be generated by unity. Then use UABEA to view the content. string m_Source is the location of the audio resource, FileSize m_Offset is the address of the beginning of the resource in the file (decimal), UInt64 m_Size is the range from the beginning to the end of the file (also decimal). Use the hex tool to extract the unity generated audio resource, put it into the game's "audio resource storage", add it to the end, and finally change the number of FileSize m_Offset to the beginning of the "add to the end" address (decimal), UInt64 m_Size is the range from the beginning to the end. address (decimal), UInt64 m_Size is changed to the size of "add to the end" (decimal!!!).
![屏幕截图 2023-10-29 105035](https://github.com/nesrak1/UABEA/assets/142726636/d1bbec02-4895-4c83-93be-2c067b990560)

**Comment by Hexaae** [comment]
**EASIER SOLUTION FOUND TO REPLACE AUDIO RESOURCES**
I successfully replaced multiple music and sounds in an unofficial patch for [X-Out Resurfaced](https://steamcommunity.com/app/3047070/discussions/0/677354977729203630/) in a much easier way than that found online and on YT involving creating a new fake Unity whole scene with the new sounds using Unity Editor (way to long and complicated requiring Unity editor installed!)...

In brief, my method consists in using free GitHub tool "oggvorbis2fsb5.exe" to convert .ogg sounds/musics into FSB5 sound banks.
Then just join together multiple FSB5 sound banks in a single file, for example "patch.resource".
Then modify as shown above in the pic by @eternalfuture-e38299 [those "StreamedResource m_Resource" fields](https://github.com/nesrak1/UABEA/discussions/281#discussioncomment-7413983) pointing them to the new file "patch.resource" (i.e. replace "resources.resource"): **m_Offset** is the starting point of each "FSB..." string (use an hex-editor to join your new FSB5 files together into "patch.resource" and inspect each bank starting value), and **m_Size** will be the actual bytes size of every single "FSB..." bank in your patch.resource.
You can (optionally?) also change the **m_Length** value (in seconds) accordingly to your new sounds sample/music, but it doesn't seem to change anything from in-game tests even if you leave wrong values... maybe is for looping sounds, I'm not sure.

Sharing this for common knowledge. Hope will help other modders too.
